include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -417,6 +444,43 @@ struct RunOptions : detail::Base<OrtRunOptions> {
   RunOptions& UnsetTerminate();
 };
 
+
+namespace detail {
+// Utility function that returns a SessionOption config entry key for a specific custom operator.
+// Ex: custom_op.[custom_op_name].[config]
+std::string MakeCustomOpConfigEntryKey(const char* custom_op_name, const char* config);
+}  // namespace detail
+
+/// <summary>
+/// Class that represents session configuration entries for one or more custom operators.
+///
+/// Example:
+///   Ort::CustomOpConfigs op_configs;
+///   op_configs.AddConfig("my_custom_op", "device_type", "CPU");
+///
+/// Passed to Ort::SessionOptions::RegisterCustomOpLibrary.
+/// </summary>
+struct CustomOpConfigs {
+  CustomOpConfigs() = default;
+  ~CustomOpConfigs() = default;
+  CustomOpConfigs(const CustomOpConfigs&) = default;
+  CustomOpConfigs& operator=(const CustomOpConfigs&) = default;
+  CustomOpConfigs(CustomOpConfigs&& o) = default;
+  CustomOpConfigs& operator=(CustomOpConfigs&& o) = default;
+
+  // Add a configuration entry for a specific custom operator. Returns a reference to *this to allow chaining calls.
+  CustomOpConfigs& AddConfig(const char* custom_op_name, const char* config_key, const char* config_value);
```

Comment:
> custom_op_name [](http://example.com/codeflow?start=40&length=15)

I am a bit torn between two alternatives. One is to bind custom_op_name to an instance  of `CustomOpConfigs`  in the __ctor and then pass only key/value pairs.

Reply:
I'm also undecided. A custom op library may contain more than one custom op, so binding an instance of `CustomOpConfigs` to a single custom op would necessitate updating `Ort::SessionOptions::RegisterCustomOpLibrary` to take in an array (or map) of custom op configs.  I'm not sure that is better.

```c++
// Current:
Ort::CustomOpConfigs op_configs;
op_configs.AddConfig("my_op", "device_type", "CPU");
op_configs.AddConfig("my_other_op", "key", "val");

session_options.RegisterCustomOpLibrary("my_ops.dll", op_configs);

// Alternative:
std::vector<Ort::CustomOpConfigs> op_configs = {{"my_op"}, {"my_other_op"}};  // Could also be unordered_map keyed by op name.
op_configs[0].AddConfig("device_type", "CPU");
op_configs[1].AddConfig("key", "val");

session_options.RegisterCustomOpLibrary("my_ops.dll", op_configs);
```